builtin/credential/github/path_login.go

```diff
@@ -203,16 +217,31 @@ func (b *backend) verifyCredentials(ctx context.Context, req *logical.Request, t
 		orgOpt.Page = resp.NextPage
 	}
 
+	orgLoginName := ""
 	for _, o := range allOrgs {
-		if strings.EqualFold(*o.Login, config.Organization) {
+		if *o.ID == config.OrganizationID {
 			org = o
+			orgLoginName = *o.Login
 			break
 		}
 	}
 	if org == nil {
 		return nil, logical.ErrorResponse("user is not part of required org"), nil
 	}
 
+	if orgLoginName != config.Organization {
```

Comment:
My thinking here was once we have the ID we will always use that and only that to verify. So if the name changes but the ID matches we will update the name. This will allow for orgs to change the name without having to update the vault config.

Are you wanting to verify with both the name and the ID?

Reply:
I see. I agree with using/relying the org ID as the verification input now that we have that available. How about emitting a warning here instead of updating the organization name, since this is being called during the login process. That way it's less magic and it's an explicit action that needs to be made by the user on the config. It also reduces the number of writes on login to just one (for the `setOrganizationID` call right above) if that's necessary.

@mickael-hc would love to get your thoughts on this as well.

Reply:
Yes, a warning probably makes more sense.

Reply:
I agree with @calvn , if the name isn't strictly necessary to function, warning here and no action seems like a good idea.